crates/pathfinder/src/storage/contract.rs

```diff
@@ -120,15 +147,64 @@ impl ContractsTable {
 
         Ok(Some(ContractCode { bytecode, abi }))
     }
+}
+
+/// Stores the mapping from StarkNet contract [address](ContractAddress) to [hash](ContractHash).
+pub struct ContractAddressesTable {}
```

Comment:
I would suggest instead of `ContractAddressesTable` and `ContractsTable` we invert it and have `ContractsTable` (address + hashes) and `ContractCodeTable` (hash, abi, code, definition).